### PR TITLE
[SPL] Aggressive cache hinting

### PR DESCRIPTION
**Pull request by watt**
targeting #420

### CODE REVIEW

File: BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift [hunk 1]
```diff
@@ -55,38 +55,136 @@ final class SPCacheTree<Key, Value, SubcacheKey> where Key: Hashable, SubcacheKe
...
     func get(key: Key, or create: (Key) -> Value) -> Value {
         if let size = values[key] {
             Logger.logCacheHit(object: self, description: path, constraint: key)
             return size
         }
         
+        // TODO:
+        // This has a negative perf impact on deep stack tests. Need to determine what the impact
+        // is on a wider variety of tests, and if there is a way we can avoid this for cases
+        // where it doesn't help (e.g. not in a scroll view).
```
Comment (n8chur) [inline BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift:95]: Does this TODO need to be updated? It looks like we addressed this by introducing an option that's only true during layout.

File: BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift [hunk 2]
```diff
@@ -55,38 +55,136 @@ final class SPCacheTree<Key, Value, SubcacheKey> where Key: Hashable, SubcacheKe
...
+        
         Logger.logCacheMiss(object: self, description: path, constraint: key)
         
         Logger.logMeasureStart(object: self, description: path, constraint: key)
         let size = create(key)
         Logger.logMeasureEnd(object: self)
 
         values[key] = size
+        
+        if options.hintRangeBoundaries {
```
Comment (n8chur) [inline BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift:154]: Do we anticipate a use case for disabling this? Are we thinking about feature flagging this?

**Review by n8chur — APPROVED** [review]

File: BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift (hunk 1 above)
Comment (n8chur) [reply BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift:95]: Er, hm, maybe we do still want to think through scoping this further, I'm not entirely sure.

Comment (watt) [reply BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift:95]: It's not yet addressed for the condition of "are we in a scroll view or other layout that measures unconstrained", which could be important.

File: BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift (hunk 2 above)
Comment (watt) [reply BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift:154]: The case I had in mind was to be able to run performance tests with it turned on & off.

**Pull request merged**